can/interfaces/socketcand/socketcand.py

```diff
@@ -40,11 +45,15 @@ def convert_can_message_to_ascii_message(can_message: can.Message) -> str:
     # Note: socketcan bus adds extended flag, remote_frame_flag & error_flag to id
     # not sure if that is necessary here
     can_id = can_message.arbitration_id
+    if can_message.is_extended_id or (can_id > 0x7FF):
```

Comment:
I'm sorry but this check for `(can_id > 0x7FF)` is still wrong.

If you want to create robust code you might mask the can_id after the check for `is_extended_id`.
(masking ext ID with & 1FFFFFFF and a 11 bit ID with & 0x7FF)

Reply:
Absolutely correct .. fixed and force pushed.

Reply:
I added Message.is_rx attribute in the convert_ascii_message_to_can_message function, instead of after it is called.